src/components/app/routes/createAppRouter.jsx

```diff
@@ -75,6 +75,22 @@ export default function createAppRouter(queryClient) {
               />
             )}
           />
+          <Route
+            path="program/:programUUID/progress"
+            lazy={async () => {
+              const { ProgramProgressPage, makeProgramProgressLoader } = await import('../../program-progress');
+              return {
+                Component: ProgramProgressPage,
+                loader: makeProgramProgressLoader(queryClient),
+              };
+            }}
```

Comment:
There were a couple ways I tried to implement this, but this is the combination that works that allows the program progress page to load but the program page does not load. Will tackle in the WIP program page PR upcoming after this PR.

```jsx
          <Route
            path="program/:programUUID"
            lazy={() => {
              const { ProgramPage } = import('../../program');
              return {
                Component: ProgramPage,
              };
            }}
            errorElement={(
              <RouteErrorBoundary
                showSiteHeader={false}
                showSiteFooter={false}
              />
            )}
          >
            <Route
              path="progress"
              lazy={async () => {
                const { ProgramProgressPage, makeProgramProgressLoader } = await import('../../program-progress');
                return {
                  Component: ProgramProgressPage,
                  loader: makeProgramProgressLoader(queryClient),
                };
              }}
              errorElement={(
                <RouteErrorBoundary
                  showSiteHeader={false}
                  showSiteFooter={false}
                />
              )}
            />
          </Route>

Reply:
If the program detail page has its own `loader`, I believe the above would *always* run the program detail page loader, even for the program progress route.

By introducing an `index` route for the program detail page, you can ensure the `loader`/`Component` rendered is specific to the program detail page.

For example, in your implementation, if the `ProgramPage` happened to render `Outlet` from `react-router-dom` in the future, it'd render both the program detail _and_ the program progress pages since both routes would be rendered simultaneously.

Something along the lines of the following:

```jsx
<Route
  path="program/:programUUID/*"
  element={<Outlet />}
  errorElement={(
    <RouteErrorBoundary
      showSiteHeader={false}
      showSiteFooter={false}
    />
  )}
 >
  <Route
    index
    lazy={() => {
      const { ProgramPage, makeProgramPageLoader } = import('../../program');
      return {
        Component: ProgramPage,
        loader: makeProgramPageLoader(queryClient),
      };
    }}
  />
  <Route
    path="progress"
    lazy={async () => {
      const { ProgramProgressPage, makeProgramProgressLoader } = await import('../../program-progress');
      return {
        Component: ProgramProgressPage,
        loader: makeProgramProgressLoader(queryClient),
      };
    }}
  />
</Route>
```

Reply:
Implemented 👍🏽